### PR TITLE
Lambda to Compehend using NodeJs for Language detection

### PR DESCRIPTION
**Pull request by abilashkumar**
*Description of changes:*
This pattern contains the source code (in AWS Javascript v3 SDK) and supporting files for a serverless application that you can deploy with the SAM CLI. Deploying the SAM file makes use of a Lambda function (uses Javascript SDK) that calls the Comprehend service to perform real-time detection of the dominant language of the input text. It makes use of the Nodejs runtime in lambda function.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: lambda-comprehend-language-sam-nodejs/README.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
+# Detect the Language (Dominant Language) of a text in real-time using Lambda and Amazon Comprehend through javascript
```
Comment (boyney123) [inline lambda-comprehend-language-sam-nodejs/README.md:1]:
```suggestion
# Detect the language (dominant language) of a text in real-time using Lambda and Amazon Comprehend through JavaScript
```

File: lambda-comprehend-language-sam-nodejs/example-pattern.json [hunk 2]
```diff
@@ -0,0 +1,57 @@
+{
+  "title": "Language Detection with Lambda and Comprehend using Javascript",
```
Comment (boyney123) [inline lambda-comprehend-language-sam-nodejs/example-pattern.json:2]:
```suggestion
  "title": "Language detection with Lambda and Amazon Comprehend using JavaScript",
```

File: lambda-comprehend-language-sam-nodejs/example-pattern.json [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  "description": "Create a Lambda function that calls the Comprehend service for detecting the Language (dominant language) of the provided text using Javscript v3 SDK.",
```
Comment (boyney123) [inline lambda-comprehend-language-sam-nodejs/example-pattern.json:3]:
```suggestion
  "description": "Creates a Lambda function that calls the Comprehend service for detecting the language of the provided text using JavaScript.",
```

File: lambda-comprehend-language-sam-nodejs/example-pattern.json [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  "level": "200",
+  "framework": "SAM",
+  "services": {
+    "from": "Lambda",
+    "to": "Comprehend"
+  },
+  "introBox": {
+    "headline": "How it works",
+    "text": [
+      "This pattern shows how to deploy a SAM template that builds a Lambda function which calls Amazon Comprehend for detecting the dominant language in a text using AWS Javascript v3 SDK."
```
Comment (boyney123) [inline lambda-comprehend-language-sam-nodejs/example-pattern.json:14]:
```suggestion
      "This pattern uses AWS SAM to build a Lambda function which calls Amazon Comprehend for detecting the dominant language in the provided text."
```

File: lambda-comprehend-language-sam-nodejs/example-pattern.json [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+      ]
+  },
+  "deploy": {
+    "text": [
+      "sam deploy"
+    ]
+  },
+  "testing": {
+    "text": [
+      "See the Github repo for detailed testing instructions."
```
Comment (boyney123) [inline lambda-comprehend-language-sam-nodejs/example-pattern.json:41]:
```suggestion
      "See the GitHub repo for detailed testing instructions."
```

File: lambda-comprehend-language-sam-nodejs/template.yaml [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+      Architectures:
+        - arm64
+      Policies:
+        - Version: '2012-10-17'
+          Statement:
+          - Sid: LambdaDetectDominantLanguage
+            Effect: Allow
+            Action:
+            - comprehend:DetectDominantLanguage
+            Resource: '*'
```
Comment (boyney123) [inline lambda-comprehend-language-sam-nodejs/template.yaml:22]: Thanks for the pattern @abilashkumar 

Is this * resource required? Can we lock this down at all?

Comment (abilashkumar) [reply lambda-comprehend-language-sam-nodejs/template.yaml:22]: @boyney123 This is required as the permission here is not tied to a resource, but in general to perform language detection on free text provided as an input. The "*" is required (it does not involve any customer resource or their saved configurations). A similar pattern published [earlier](https://github.com/aws-samples/serverless-patterns/blob/main/lambda-translator/template.yaml#L20).

**Comment by jbesw** [comment]
Due to the large amount of custom code, this should be submitted as a repo, not a pattern: https://serverlessland.com/repos/.

**Pull request closed without merging**